src/target/target.cc

```diff
@@ -456,6 +478,11 @@ void TargetInternal::ConstructorDispatcher(TVMArgs args, TVMRetValue* rv) {
                  << runtime::ArgTypeCode2Str(arg.type_code());
     }
     return;
+  } else if (args.num_args == 2) {
+    const auto &argt = args[0], &argh = args[1];
+    auto func = PackedFunc(ConstructorDispatcher);
+    *rv = Target(func(argt).AsObjectRef<Target>(), func(argh).AsObjectRef<Target>());
```

Comment:
We only care about the case that both arguments are "Target"

```suggestion
    if (args[0]->IsObjectRef<Target>() && args[1]->IsObjectRef<Target>()) {
      Target target = args[0];
      Target host = args[1];
      *rv = Target(target, host);
    }
```